upb/def.c

```diff
@@ -2659,7 +2812,7 @@ static void create_msgdef(symtab_addctx *ctx, const char *prefix,
   m->nested_ext_count = n;
   m->nested_exts = symtab_alloc(ctx, sizeof(*m->nested_exts) * n);
   for (i = 0; i < n; i++) {
-    create_fielddef(ctx, m->full_name, NULL, fields[i], &m->nested_exts[i]);
+    create_fielddef(ctx, m->full_name, m, fields[i], &m->nested_exts[i], true);
     ((upb_fielddef*)&m->nested_exts[i])->index_ = i;
   }
```

Comment:
this method has gotten a bit too long for my taste.  Can you break it into some well named helpers?

Reply:
Done.